src/main/java/com/google/sps/data/EventUpdater.java

```diff
@@ -36,7 +49,159 @@ public Entity getEntity() {
     return this.entity;
   }
 
+  // TODO: When editing event (after event has been registered), we must check if the user sending the request has the right credentials - add GivrUser.isModeratorOfOrgWithId(long organizationId)
   public void updateEvent(HttpServletRequest request, GivrUser user, boolean forRegistration, EmbeddedEntity historyUpdate) throws IllegalArgumentException {
-    
+    HashMap<String, String> formProperties = new HashMap<String, String>();
+
+    // Format is (Form EntryName, Entity PropertyName)
+    formProperties.put("event-primary-organization-id", "eventOwnerOrgId");
+    formProperties.put("event-name", "eventName");
+    formProperties.put("event-partner", "eventPartnerNames");
+    formProperties.put("event-details", "eventDetails");
+    formProperties.put("event-contact-email", "eventContactEmail");
+    formProperties.put("event-contact-phone-num", "eventContactPhone");
+    formProperties.put("event-contact-name", "eventContactName");
+    formProperties.put("event-street-address", "eventStreetAddress");
+    formProperties.put("event-city", "eventCity");
+    formProperties.put("event-state", "eventState");
+    formProperties.put("event-zip-code", "eventZipcode");
+
+    /* Optional Properties can be left blank in the request form.
+     *
+     * The following properties are optional:
+     * - eventPartnerNames
+     * - eventDetails
+     */
+    Set<String> optionalProperties = new HashSet<String>();
+    optionalProperties.add("eventPartnerNames");
+    optionalProperties.add("eventDetails");
+
+    long ownerOrgId = 0;
+
+    for (Map.Entry<String, String> entry: formProperties.entrySet()) {
+      String propertyKey = entry.getValue();
+      String formKey = entry.getKey();
+      String formValue = "";
+
+      if (optionalProperties.contains(propertyKey)) {
+        formValue = request.getParameter(formKey) == null ? "" : request.getParameter(formKey);
+      } else {
+        try {
+          formValue = getParameterOrThrow(request, formKey);
+        } catch (IllegalArgumentException err) {
+          logger.log(Level.SEVERE, "Form value for: " + propertyKey + " cannot be left blank.");
```

Comment:
should you re-raise the exception here? Alternatively, don't handle the exception in this function and let the servlet handle it.